{james wip}

codemonkeman opened the pull request:
Users can now register for parties. This will need some polish, but that's part of usability. It works. 
Closes #37

codemonkeman committed:
SearchUser now works, functionally the same as parties, had some
weird issues to work out, but now it even has the avatar picture

codemonkeman committed:
join party functionality now works, needs polish, but it works

ThaddG commented on frontend/src/components/SearchUser/SearchUser.js:
```diff
@@ -1,23 +1,75 @@
+  const [search, setSearch] = useState('');
+
+
+  console.log('whatever in a string', allUsers);
```
That console.log isn't needed anymore.

ThaddG commented on frontend/src/components/SearchUser/SearchUser.js:
```diff
@@ -1,23 +1,75 @@
+  const { REACT_APP_URL } = process.env;
   const [user, setUser] = useContext(UserContext);
+  const [allUsers, setAllUsers] = useState([]);
   const [homeRender, setHomeRender] = useContext(HomeRenderContext);
```
Don't need homeRender here. That's only for Home.js

ThaddG commented on frontend/src/components/SearchUser/SearchUser.js:
```diff
@@ -1,23 +1,75 @@
 
+  useEffect(()=>{
+    getAllUsers();
+  } , [homeRender])
```
This doesn't run, can be removed.

ThaddG commented on frontend/src/components/ViewParty/ViewParty.js:
```diff
@@ -10,6 +10,28 @@ import axios from 'axios';
+      },
+    };
+  
+  const Link = `${process.env.REACT_APP_URL}/Party/${props.ID}/Update`;
```
That link has one too many '/'s. replace it with this:
const Link = `${process.env.REACT_APP_URL}Party/${props.ID}/Update`;

ThaddG commented on frontend/src/components/ViewParty/ViewParty.js:
```diff
@@ -10,6 +10,28 @@ import axios from 'axios';
+  const Link = `${process.env.REACT_APP_URL}/Party/${props.ID}/Update`;
+
+  const payload={
+    Attendees: props.Attendees.push({ID:props.user.ID,Username: props.user.Username})
```
I tested this out and it returns an integer. So to fix, declare an array above it, push/concat to it then let the value of the key be the declared array.

codemonkeman committed:
Fixed the problems with the join party feature.
cleaned up information for the party views on home.js and the table on
ViewParty is now dynamic,

codemonkeman committed:
Merge branch 'beta' into {JamesWIP}

codemonkeman committed:
fixed join party button

ga2599 commented:
great work
